Added a new doc for covert from k8s deploy to knative service (#3859)
knative/docs: docs/serving/convert-deployment-to-knative-service.md

> # Covert Kubernetes Deployment to Knative Service

Author: Done

> Learn how to covert Kubernetes Deployment to Knative Service.

Reviewer: ```suggestion
This topic shows how to convert a Kubernetes Deployment to a Knative Service.
```

Author: Done

> ## Determining if your workload is a good fit for Knative

Reviewer: ```suggestion
## Determine if your workload is a good fit for Knative
```

Author: Done

> In general, if your Kubernetes workload is a good fit for Knative, you should be able to remove a lot of your manifest to create a Knative Service.

Reviewer: ```suggestion
In general, if your Kubernetes workload is a good fit for Knative, you can remove a lot of your manifest to create a Knative Service.
```

Author: Done

There are three aspects you need to consider:

> - All work done is triggered by HTTP

Reviewer: ```suggestion
- All work done is triggered by HTTP.
```

Author: Done

> - The container is stateless (all state is stored elsewhere or can be re-created)

Reviewer: ```suggestion
- The container is stateless. All state is stored elsewhere or can be re-created.
```

Author: Done

> - Uses only Secret and ConfigMap volumes

Author: Done

> - Your workload uses only Secret and ConfigMap volumes.

Reviewer 2: I think this should also include autoscaling. The shown transformation would include scale the nginx pod for roughly every 100 concurrent HTTP requests, so you're saving even more YAML :)

Author: Good comment, thanks @markusthoemmes .

How about add a new section named as follow:

## Benefits

- Reduce footprint of the service instance as the instance will be scale to 0 when it became idle.
- Performance can be improved via build-in autoscaling for knative service.

Thoughts?

## Example conversion

> The following will use an Kubernetes Nginx Deployment and Service, and covert it to Knative Service.

Reviewer: ```suggestion
The following example uses a Kubernetes Nginx Deployment and Service, and converts it to a Knative Service.
```

Author: Done

> The following example shows a Kubernetes Nginx Deployment and Service, and shows how it converts to a Knative Service.

Reviewer 3: Reference the k8s docs this was taken from?

Author: @evankanderson Done, thanks!
